### PR TITLE
[cartservice] Update to .NET 7

### PR DESCRIPTION
**Pull request by Kielek**
# Changes

Update target and docker images to .NET7 for cartservice.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* ~~[ ] Appropriate documentation updates in the [docs][]~~ N/A
* ~~[  ] Appropriate Helm chart updates in the [helm-charts][]~~ N/A

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Comment by Kielek** [comment]
@cartersocha, I was able to reproduce. I am not sure how I missed that while testing (probably I have made mistake while building new docker image).

Bug reported under https://github.com/open-telemetry/opentelemetry-dotnet/issues/4600

Locally, I have removed Trimming option. It will increase image size, but it is working correctly.

**Review by cartersocha — APPROVED** [review]
It's working now thanks!

File: src/cartservice/src/Dockerfile [hunk 1]
```diff
@@ -26,12 +26,12 @@ RUN \
   RUNTIME_IDENTIIFER=linux-musl-x64; \
   if [ "$(uname -m)" = "aarch64" ]; then RUNTIME_IDENTIIFER=linux-musl-arm64; fi; \
   dotnet restore ./src/cartservice.csproj -r $RUNTIME_IDENTIIFER && \
-  dotnet publish ./src/cartservice.csproj -p:PublishSingleFile=true -r $RUNTIME_IDENTIIFER --self-contained true -p:PublishTrimmed=True -p:TrimMode=Link -c Release -o /cartservice --no-restore
+  dotnet publish ./src/cartservice.csproj -p:PublishSingleFile=true -r $RUNTIME_IDENTIIFER --self-contained true -p:PublishTrimmed=False -p:TrimMode=Link -c Release -o /cartservice --no-restore
```
Comment (pellared) [inline src/cartservice/src/Dockerfile:29]: Could you explain the change from `-p:PublishTrimmed=True` to `-p:PublishTrimmed=False`?

Comment (pellared) [reply src/cartservice/src/Dockerfile:29]: Nevermind. It is here: https://github.com/open-telemetry/opentelemetry-demo/pull/942#issuecomment-1594178161

**Review by pellared — APPROVED** [review]

**Comment by Kielek** [comment]
@cartersocha, trimming was working on .NET 6 by accident (it is producing warnings). There is a plan to make a support for trimming operation in OpenTelemetry .NET for 1.6 version (tentatively scheduled on November). 

It should be good to merge for now.

**Pull request merged**